server/leafnode.go

```diff
@@ -1385,7 +1430,7 @@ func (c *client) processLeafSub(argo []byte) (err error) {
 			checkPerms = false
 		}
 	}
-	if checkPerms && !c.canExport(string(sub.subject)) {
+	if checkPerms && c.isHubLeafNode() && !c.canSubscribe(string(sub.subject)) {
```

Comment:
With this change it means that only LN that have remotes (where you can define permissions) and that act as a Hub will apply permissions. The LN that used to have remote with deny permission but be a spoke will now freely send messages over..

Reply:
Will send the subscriptions yes.

Reply:
If we allow export on "foo", and remote hub has interest in "*", the old way would not work and let "foo" flow through the hub.

Reply:
If this server had a deny export of "foo", and received a subscription request on "*" it would accept the subscription on "*" but the deliverMsg pubAllowed check would suppress messages on "foo" (but allow messages on "bar"). If it had received a subscription request on "foo", it would have rejected the subscription (and closed the connection). I have just tested that and it worked as I describe here.

Reply:
Note that with your changes, the server that has a deny export "foo" is now accepting that subscription. It seems that we still suppress on publish, so that's ok, but we should not even accept the subscription on "foo" if we can't export on it.

Reply:
Even worse, if the remote is now "hub" then messages on "foo" are sent!

Reply:
After some discussion it seems that we need to remove in deliverMsg() the check for isSpokeLeafNode() and simply check if leaf kind.

Reply:
Yes agree, just made that change.